test/new-e2e/cws/e2e_cws_test.go

```diff
@@ -0,0 +1,150 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package cws
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	cws "github.com/DataDog/datadog-agent/test/new-e2e/cws/lib"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner/parameters"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/params"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2params"
+)
+
+type agentSuite struct {
+	e2e.Suite[e2e.AgentEnv]
+	apiClient     cws.MyAPIClient
+	signalRuleID  string
+	agentRuleID   string
+	dirname       string
+	filename      string
+	testID        string
+	desc          string
+	agentRuleName string
+	policies      string
+}
+
+func TestAgentSuite(t *testing.T) {
+	agentConfig := "hostname: momar-e2e-test-host\nlog_level: DEBUG\nlogs_enabled: true"
+	e2e.Run(t, &agentSuite{}, e2e.AgentStackDef(
+		[]ec2params.Option{
+			ec2params.WithName("cws-e2e-tests"),
+		},
+		agentparams.WithAgentConfig(agentConfig),
+		agentparams.WithSecurityAgentConfig("runtime_security_config:\n  enabled: true"),
+		agentparams.WithSystemProbeConfig("system_probe_config:\n  enabled: true\n  log_level: trace\nruntime_security_config:\n  enabled: true\nlog_patterns:\n  - module.APIServer.*"),
+		agentparams.WithVersion("7.46.0"),
+	), params.WithDevMode())
+}
+
+func (a *agentSuite) SetupSuite() {
+	// Create temporary directory
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
You could do this setup within pulumi context using the [filemanager](https://github.com/DataDog/datadog-agent/blob/main/test/new-e2e/examples/customenv_with_filemanager_test.go), with a custom environment. 

The advantage is that pulumi would keep track of deleting the directory on destroy. On `TearDownSuite` and for `devMode` on, you would need to clear what you add afterwards, so files in it, the disadvantage is that it is not intuitive to pass the temp dir name out of pulumi context. You'd need to pass it with the environment, this could be a feature request.

Reply:
I thought of using the filemanager, but yeah I would need to define a custom environment, which I'm trying to avoid for the host-based agent. I'll leverage the filemanager when I implement the tests for containers and kubernetes environments.